design/unnumbered-bgp.md

```diff
@@ -0,0 +1,541 @@
+# MetalLB support for BGP Unnumbered
+
+## Summary
+
+BGP Unnumbered allows routers to peer with each other when direct connected
+without the need for IPs (the BGP configuration uses interface names). Once
+peering takes places, exchange IPv4 or IPv6 prefixes can take place. That
+feature can be used by MetalLB to simplify configuration both on the network
+fabric and on the cluster side.
+
+## Motivation
+
+Using BGP unnumbered peering, which dynamically discovers IPV6 neighbors,
+reduces the burden of the network administrator to configue all
+interfaces on the network fabric or on the cluster nodes to have IPv4
+addressing just for the BGP peering. By using BGP unnumbered, cluster administrator
+avoids to specify the address of each neihbor. Unnumbered BGP utilizes IPv6
+link local address to automatically decide which peer to connect to.
+
+## Limitation/Requirements
+
+- To use unnumbered BGP feature, there must be a link (point-to-point, directly
+  connected) between the interfaces two BGP speakers are using
+- IPv6 must be enabled on that interfaces, and with LLA (Link Local IPv6 Address) enabled.
+  Note: the use of IPv6 LLA does not require a full IPv6 deployment
+- IPv6 RAs (Router Advertisments) must be enabled (it should be safe when p2p and no IP address on the interface).
+
+## Goals
+
+- Make the BGP peering from Server to the ToR (TopOfRack) easier to configure
+  and have the same configuration for all servers (assuming the interfaces' names
+  are the same across the cluster nodes)
+- Remove the task of network admins to setup up addressing on those interfaces
+  if it is needed only for BGP peering
+
+## Non Goals
+
+- Make it work over no point-to-point connection
+- Define/Discuss dual ToR architecture, on top of bond interfaces or other HA strategies
+- [Dynamic BGP Peering](https://ipwithease.com/dynamic-bgp-peering/) is a
+  feature different than unnumbered BGP and not supported by Metallb. Both
+  feature reduce the configuration needed for the BGP peering. The main difference is
+  that Dynamic BGP peering still needs IP addresses and does not require point-to-point links.
+- Support `remote-as` to accept `<external|internal|auto>`
+
+## User Stories
+
+### Story 1
+
+As cluster administrator, I want to configure metallb without configuring IP addresses.
+
+### Story 2
+
+As network administrator, I want to avoid configuring IP addresses for
+point-to-point connection just for the BPG peering reason.
+
+### Story 3
+
+As network administrator, I want to reduce the size of the configuration in the ToR router.
+
+## The API Changes
+
+### Option 2. Add a new Interface field
+
+We can introduce a new field `Interface string` which if defined then Unnumber
+BGP takes place. In that case, the address must NOT have value. For
+completeness, we could use field might be used to support `LLA%zone` [user case](https://github.com/metallb/metallb/issues/2423).
+So at the end, the API documentation will look like
```

Comment:
as the other comment, I think this could be a lot shorter (no need to specify everything and its consequence)

Reply:
I agree, but I prefer to remove once @fedepaol take another feedback and he is ok with the consequence or not (we could decide no to support the LLA)

Reply:
Let's focus on unnumbered and maybe implement LLA afterwards if we need / want to.
What's important here is that the API is compatible with that. If we want to mention it, let's add a subsection in the bottom

Reply:
ok, removed the reference to LLA